### PR TITLE
feat: add Reactor support

### LINKED ISSUE #627: Support for Reactor library
Feature request - low urgency but would be useful to avoid reimplementing the code externally or bridging via RxJava2.  This is a fairly well adopted library that is an alternative to RxJava.

http://projectreactor.io/
https://github.com/reactor/reactor-core

### PR DESCRIPTION
**Pull request by aoudiamoncef**
resolves #627

Hello @martinbonnin,

I've added Reactor support for non Android projects.

I'm working on Android integration based on [RxAndroid](https://github.com/ReactiveX/RxAndroid) to get Android aware schedulers.

Thanks

### CODE REVIEW

File: buildSrc/build.gradle.kts [hunk 1]
```diff
@@ -10,6 +10,7 @@ repositories {
   gradlePluginPortal()
   google()
   mavenCentral()
+  mavenLocal()
```
Comment (martinbonnin) [inline buildSrc/build.gradle.kts:13]:
```suggestion
```

**Review by martinbonnin — APPROVED** [review]
LGTM!

**Comment by aoudiamoncef** [comment]
Hi @martinbonnin,

LGTM too

**Comment by martinbonnin** [comment]
CI failed on metalava signatures. For some reason they need to be generated with JDK8. I just pushed an update

**Comment by martinbonnin** [comment]
Thanks for the contribution!

**Comment by aoudiamoncef** [comment]
@martinbonnin 

Please, could you add missing Reactor documentation url to Readme.md 

[Reactor](https://www.apollographql.com/docs/android/advanced/reactor)

thanks

**Comment by martinbonnin** [comment]
PR for adding the doc topic: https://github.com/apollographql/apollo-android/pull/3141

**Pull request merged**